src/L6/datasorter.py

```diff
@@ -57,14 +58,27 @@ def set_input_data(self, file_path_name: str):
                 for i in row:
                     if i.isdigit():
                         self.data.append(float(i))
+                    elif re.match("^\d+?\.\d+?$", i):
+                        self.data.append(float(i))
                     else:
                         raise ValueError("The CSV contains non-numeric values")
 
         return True
 
     # Exercise 26
     def set_output_data(self, file_path_name: str):
-        pass
+        if not file_path_name.lower().endswith(".csv"):
+            file_path_name = file_path_name + ".csv"
+        if len(self.data) == 0:
+            raise EmptyDataArrayException
+        with open(file_path_name, 'w+') as file:
+            output = ""
+            for item in self.data:
+                output = output + str(item) + ", "
```

Comment:
Don't output strings directly to a CSV file, if you have `csv` already imported, use it to write to the file.
More info here https://docs.python.org/3/library/csv.html#csv.writer
And here https://realpython.com/python-csv/